### ISSUE TITLE
static arrays (db, has_one, has_many, etc...) should be protected not private

### ISSUE THREAD

**Issue by tony13tv** [silverstripe/silverstripe-framework#3999]
I think static arrays such as `$db`, `$has_one`, `$has_many`, etc... should be protected not private, since inheritance is supposed to work this way.

**Comment by dhensby** [comment]
@tony13tv thanks for the feedback.

The `private statics` have a special purpose in SilverStripe and so their visibility is really not subject of a conventional argument about inheritance. They are practically reserved for use of the `Config` system to parse out because you're right that typically statics shouldn't be private in this way.

By choosing `private static` we are allowing people to define `protected statics` without polluting the `Config` system.

In all honesty, you should be using YAML to define these vars if you want to be purist about it.

In summary, there's basically no way I can see the `Config` statics being made anything other than `private` - it's more likely they'll just be removed and defined in YAML instead
